pkg/kubelet/prober/worker.go

```diff
@@ -222,6 +222,22 @@ func (w *worker) doProbe() (keepGoing bool) {
 			w.pod.Spec.RestartPolicy != v1.RestartPolicyNever
 	}
 
+	// Graceful shutdown of the pod.
+	if w.pod.ObjectMeta.DeletionTimestamp != nil {
```

Comment:
We don't need to do any container checking after we see the pod deletion timestamp is set so, in this case, should we not run this logic before we do any container checking? (i.e. move this addition above L191)

Reply:
well I need to have a valid `w.containerID` to set the probe results, don't I?

Reply:
Ahh yes sure. so maybe before the `onHold` check?

Reply:
(I guess it doesn't really matter as much, this is nonblocking)

Reply:
Yeah, in fact we probably have more often `c.State.Running == nil` than `w.pod.ObjectMeta.DeletionTimestamp != nil`.